### ISSUE TITLE
Found '⁨{{FrontSide}}⁩', but there is no field called 'FrontSide'

### ISSUE THREAD

**Issue by Selduro** [ankidroid/Anki-Android#9049]
The following issue only occured after updating to 2.15. Via Fdroid I reverted back to 2.14.6 and the problem does not occur there.

A shared deck I am using is utilizing custom templates with (of course) custom fields.
There is no problem when reviewing them on ankiweb itself via browser or in the desktop app (kubuntu version 2.1.44).

When trying to review the same card in ankidroid I get the error message `Found '⁨{{FrontSide}}⁩', but there is no field called 'FrontSide'` where the front side of the card is supposed to be. When pressing the corresponding button the back side appears normally and I am able to rate my memorization.

This is the template for the front side (of a random note using that template, not all available fields are in use):

<details>
  <summary>frontside</summary>
 
  
```
{{#Frage}}{{#Antwort}}

<script>qFade=0; if (typeof anki !== 'undefined') anki.qFade=qFade;</script>

<div id="UeberAnzeigen" class="UeberAnzeigen"  style="position:absolute;right:0px;top:0px;cursor: pointer;" onclick="if (document.getElementById('Karte').style.display=='') { document.getElementById('Karte').style.display='none';document.getElementById('Ueber').style.display='';  document.getElementById('UeberAnzeigen').className='UeberAusblenden';} else {document.getElementById('Karte').style.display='';document.getElementById('Ueber').style.display='none'; document.getElementById('UeberAnzeigen').className='UeberAnzeigen';}"></div>

<div id=Ueber align=center style="display:none;">
<img src="_Logo.png">
<h1>Jura-Vorlagen <span style="font-weight:normal">für Anki</span></h1>
<p>Thomas Kahn</p>
<p><a href="http://thomaskahn.de/jura-vorlagen/">thomaskahn.de/jura-vorlagen/</a></p>
</div>

<div id=Karte>

<center>

<table>
	<tr>
		<td id=Frage>{{Frage}}</td>
	</tr>
</table>

</center>

<p>

{{#Merkhilfe}}<div class=zusatzinfo><a class=standardlink href="#" onclick="document.getElementById('Merkhilfe2').style.display='block';return false;">Merkhilfe</a><span id="Merkhilfe2" class=andere_karte_dieser_notiz style="display: none;">{{Merkhilfe}}</div>{{/Merkhilfe}}

{{#Tags}}
	<div class=zusatzinfo>
		<a class=standardlink href="#" onclick="document.getElementById('tags').style.display='block';return false;">Tags</a>
	</div>
	<div id="tags" style="display: none;" class=zusatzinfo>
		{{Tags}}
	</div>
{{/Tags}}

</p>

<script>
function aufdecken_frageseite() {
	var ausklappbare_elemente = document.getElementsByClassName("andere_karte_dieser_notiz");
	for (var i in ausklappbare_elemente) {
	  try {
		ausklappbare_elemente[i].style.display = "block";
	  } catch (err) {
	  }
	}
}

//Alles aufdecken, wenn man Shift + Leertaste drückt - funktioniert allerdings nur, wenn die Karte fokussiert ist und nicht die Buttons, um die Antwort zu bewerten.
document.onkeypress = function (e) {
	var evt = window.event || e;   
	if (evt.keyCode == 32 && evt.shiftKey) {
		aufdecken_frageseite()
		return false;
	}
}
</script>

<script>
function standard_gesetz_erkennen(){var deck="{{Deck}}"
if(deck.search(/Privatrecht|Zivilrecht/i)!=-1)
return"BGB"
else if(deck.search(/Strafrecht/i)!=-1)
return"StGB"
return""}
if("{{FrontSide}}"==""||"{{FrontSide}}"=="{unknown field FrontSide}"){var injectScript=(src)=>{return new Promise((resolve,reject)=>{const script=document.createElement('script');script.src=src;script.async=!0;script.onload=resolve;script.onerror=reject;document.head.appendChild(script)})};setTimeout(function(){(async()=>{if(typeof normangaben_in_allen_relevanten_elementen_verlinken==='undefined'){await injectScript('_normen_verlinken.js')}
normangaben_in_allen_relevanten_elementen_verlinken(standard_gesetz_erkennen())})()},0)}
</script>

{{/Antwort}}{{/Frage}}
```
  
</details>

and this is the template for the backside (although the back side is working fine)

<details>
  <summary>back side</summary>
  
```
<div id=frontside>{{FrontSide}}</div>

<hr id=answer>

<center>

<table>
	<tr>
		<td id=antwort>{{Antwort}}</td>
	</tr>
</table>

</center>

<p>

<div class=zusatzinfo id=feedback style="float:right">
	<a class=standardlink href="http://thomaskahn.de/basiskarten/feedback.php">Verbesserungsvorschlag</a>
</div>

{{#Begründung}}<div class=zusatzinfo><a class=standardlink href="#" onclick="document.getElementById('Begruendung').style.display='block';return false;">Begründung</a><span id="Begruendung" class=andere_karte_dieser_notiz style="display: none;">{{Begründung}}</div>{{/Begründung}}

<!--Auskommentiert:
{{#Merkhilfe}}<div class=zusatzinfo><a class=standardlink href="#" onclick="document.getElementById('Merkhilfe').style.display='block';return false;">Merkhilfe</a><span id="Merkhilfe" style="display: none;">{{Merkhilfe}}</div>{{/Merkhilfe}}-->

{{#Fachbegriff}}<div class=zusatzinfo><a class=standardlink href="#" onclick="document.getElementById('Bezeichnung').style.display='block';return false;">Fachbegriff</a><span id="Bezeichnung" class=andere_karte_dieser_notiz style="display: none;">{{Fachbegriff}}</div>{{/Fachbegriff}}

{{#Norm}}<div class=zusatzinfo><a class=standardlink href="#" onclick="document.getElementById('Norm').style.display='block';return false;">Norm</a><span id="Norm" class=andere_karte_dieser_notiz style="display: none;">{{Norm}}</div>{{/Norm}}

{{#Klausurstandort}}<div class=zusatzinfo><a class=standardlink href="#" onclick="document.getElementById('Klausurstandort').style.display='block';return false;">Klausurstandort</a><span id="Klausurstandort" class=andere_karte_dieser_notiz style="display: none;">{{Klausurstandort}}</div>{{/Klausurstandort}}

</p>

<div id="Identifikator" style="display:none">{{Frage}}</div>
<div id="tags" style="display: none;" class=zusatzinfo>{{Tags}}</div>
<div id="bk-portal-token" style="display:none"></div>
<div id="quellen" style="display:none">{{Quellen}}</div>

<script>

function _(element) {
  return document.getElementById(element);
}

//Prüfen, ob die Karte leer ist und ggf. Hinweise anzeigen

if (_("frontside").innerHTML == "") {
	var feld = "{{Card}}".split(" ")[1]
	if (feld == "Lösung")
		feld = "Antwort"
	_("antwort").innerHTML = "<center><b><span style='color: black; background-color: #ffe000; padding: 3px'>Leere Karte gefunden</span></b><br><br></center>Diese Karte (<font color=#1a8cff>{{Card}}</font>) ist leer, weil das dazugehörige Feld (<font color=#1a8cff>" +feld +"</font>) der Notiz leer ist.<br><br><i>Andere Felder derselben Notiz können natürlich ausgefüllt sein. Wenn du nicht weißt, was der Unterschied zwischen Notizen und Karten ist, <a href='http://thomaskahn.de/jura-vorlagen/notizen_karten/'>klick hier.</a><br><br>Um diese und andere <a href='http://thomaskahn.de/jura-vorlagen/leere_karten/'>leere Karten</a> zu löschen, geh ins Anki Hauptmenü und klick oben auf \"Extras\" (bzw. \"Werkzeuge\") und dann auf \"Leere Karten ...\". Die übrigen (nicht-leeren) Karten der betroffenen Notizen bleiben natürlich erhalten.</i>"
}

//Funktionen, um den Verbesserungsvorschlag-Button richtig zu verlinken:

function formatDeckStr(deck) {
	deck = deck.replace(/&/gi, '+');
	deck = deck.replace(/[`~!@#$%^&*()_|+\-=?;'",.<>]/gi, ' '); //Nur : wird nicht ersetzt.
	deck = deck.split("[").join(" ");
	deck = deck.split("]").join(" ");
	deck = deck.split("{").join(" ");
	deck = deck.split("}").join(" ");
	deck = deck.split("/").join(" ");
	deck = deck.split("\\").join(" ");
	deck = deck.replace(/[\n\r\s]+/gi, ' ');
	//deck = deck.trim();
	deck = deck.split(" ").join("_");
	deck = encodeURI(deck);
	return deck;
}

function removeSection(str, section_start, section_end) {
	var section_start_index = str.indexOf(section_start);
	while (section_start_index != -1) {
		var section_end_index = str.indexOf(section_end);
		str = str.slice(0, section_start_index) + str.slice(section_end_index +section_end.length, str.length);
		section_start_index = str.indexOf(section_start);
	}	
	return str;
}

function formatContentStr(content) {
	//Cloze-Formatierung entfernen
	var parts = content.split("<span class=\"cloze\">")
	for (i = 1; i < parts.length; i++)
		parts[i] = parts[i].replace("</span>", "")
	content = parts.join("");
	
	//Unsichtbare <style> und <head>-Sections entfernen
	content = removeSection(content, "<style", "</style>");
	content = removeSection(content, "<head", "</head>");
	
	content = content.replace(/<[^>]+>/gi, ' ')
	content = content.replace(/&amp;/gi, ' ');
	content = content.replace(/&nbsp;/gi, ' ');
	content = content.replace(/&gt;/gi, ' ');
	content = content.replace(/&lt;/gi, ' ');
	content = content.replace(/ -/gi, ' ');
	content = content.replace(/[`~!@#$%^&*()_|+\-=?;:'",.<>]/gi, ' ');
	content = content.split("[").join(" ");
	content = content.split("]").join(" ");
	content = content.split("{").join(" ");
	content = content.split("}").join(" ");
	content = content.split("/").join(" ");
	content = content.split("\\").join(" ");
	content = content.replace(/[\n\r\s]+/gi, ' ');
	content = content.trim()
	content = content.substring(0, 100);
	content = content.trim()
	content = content.split(" ").join("_");
	content = encodeURI(content);
	content = content.replace(/%0A/gi, '_');
	content = content.replace(/%09/gi, '_');
	content = content.replace(/__+/gi, '_');
	return content;
}

function get_basiskarten_id() {
	var quellen = _("quellen").innerHTML.toLowerCase();
	var start = "[basiskarten_id:"
	var start_index = quellen.indexOf(start);
	if (start_index != -1) {
		var str = quellen.slice(start_index +start.length, quellen.length);
		return str.slice(0, str.indexOf("]")).trim();
	} else
		return "";
}

function get_basiskarten_update() {
	var quellen = _("quellen").innerHTML.toLowerCase();
	var start = "[basiskarten_update:"
	var start_index = quellen.indexOf(start);
	if (start_index != -1) {
		var str = quellen.slice(start_index +start.length, quellen.length);
		return str.slice(0, str.indexOf("]")).trim();
	} else
		return "";
}

//Funktion, um zu prüfen, ob es Verbesserungsvorschläge für diese Karte gibt

function status_verbesserungsvorschlaege(nid, update, token, inhalt, output_element_id) {

	var img = document.createElement("img");
	img.id = "myImg";
	img.src = "https://basiskarten-api.anvil.app/_/api/suggestions_state_img/?" +nid +update +token +"inhalt=" +inhalt
	img.onerror = function() {
		_("myImg").style.display = "none";
	};
	  
	_(output_element_id).appendChild(img);

	function loaded() {
		var breite = _("myImg").offsetWidth;
		_("myImg").style.display = "none";
		
		if (breite == 1) {
			_(output_element_id).innerHTML = _(output_element_id).innerHTML + " <span class=sugg><font color=#DF0101>(<b>●</b>)</font></span>"			
		} else if (breite == 2) {
			_(output_element_id).innerHTML = _(output_element_id).innerHTML + " <span class=sugg><font color=green>(<b>●</b>)</font></span>"
		} else if (breite == 3) {
			_(output_element_id).innerHTML = _(output_element_id).innerHTML + " <span class=sugg>(<b>●</b>)</span>"
		} else if (breite == 0) {
			
		}
	}

	if (img.complete) {
	  loaded()
	} else {
	  img.addEventListener('load', loaded)
	}
	
	return true
}

//Prüfen: Verbesserungsvorschlag-Button anzeigen und automatische Abfrage, ob Verbesserungsvorschläge dafür vorliegen, nur bei Karten mit "Basiskarten"-Tag:

var whitelist = "basiskarten"
var tags = _("tags").innerHTML.trim().toLowerCase();	
if (tags != "" && (tags.split(" ").indexOf(whitelist) != -1 || whitelist == tags)) {
	var Deck   = formatDeckStr("{{Deck}}");
	var Karte  = formatDeckStr("{{Card}}");
	var Inhalt = formatContentStr(_("Identifikator").innerHTML);
	var nid    = "nid=" +get_basiskarten_id() +"&";
	if (nid == "nid=&")
		var nid = "";
	var update = "update=" +get_basiskarten_update() +"&";
	if (update == "update=&")
		var update = "";
	_("feedback").innerHTML = "<a id='feedback_inner' href='"+"https://basiskarten.anvil.app/#?" +nid +update +"inhalt=" +Inhalt +"&quelle=" +Karte +"&deck=" +Deck +"'>Verbesserungsvorschlag</a>";

	try {
		var token = "token=" +getComputedStyle(_("bk-portal-token"), ':after').content.trim().split("'").join("").split('"').join("") +"&";
	} catch (err) {
		var token = "";
	}
	
	status_verbesserungsvorschlaege(nid, update, token, Inhalt, "feedback_inner")
		
} else
	_("feedback").style.display = "none";

function aufdecken_antwortseite() {
	var ausklappbare_elemente = document.getElementsByClassName("andere_karte_dieser_notiz");
	for (var i in ausklappbare_elemente) {
	  try {
		ausklappbare_elemente[i].style.display = "block";
	  } catch (err) {
	  }
	}
}

//Alles aufdecken, wenn man Shift + Leertaste drückt - funktioniert allerdings nur, wenn die Karte fokussiert ist und nicht die Buttons, um die Antwort zu bewerten.
document.onkeypress = function (e) {
	var evt = window.event || e;   
	if (evt.keyCode == 32 && evt.shiftKey) {
		aufdecken_antwortseite()
		return false;
	}
}

</script>

<!--Ab hier beginnt die Druckvariante dieser Notiz. Bitte diesen Kommentar nicht entfernen, da dadurch die Bereiche voneinander getrennt werden.-->

<div class=Druck>

<h1 class=DeckAlsUeberschrift>{{Deck}}</h1>
<h1 class=TagsAlsUeberschrift>{{Tags}}</h1>

<table>
	<tr>
		<td class=Identifikator_Druck style="font-weight:bold">{{Frage}}</td>
	</tr>
	<tr>
		<td><br></td>
	</tr>
	<tr>
		<td>{{Antwort}}</td>
	</tr>
	<tr>
		<td class=trenner></td>
	</tr>{{#Begründung}}
	<tr>
		<td class=anschlussfrage><font face="Arial"> ► </font>Begründung</td>
	<tr>
		<td class=anschlussantwort>{{Begründung}}<div class=trenner></div></td>
	</tr>{{/Begründung}}{{#Merkhilfe}}
	<tr>
		<td class=anschlussfrage><font face="Arial"> ► </font>Merkhilfe</td>
	<tr>
		<td class=anschlussantwort>{{Merkhilfe}}<div class=trenner></div></td>
	</tr>{{/Merkhilfe}}{{#Fachbegriff}}
	<tr>
		<td class=anschlussfrage><font face="Arial"> ► </font>Fachbegriff</td>
	<tr>
		<td class=anschlussantwort>
		{{Fachbegriff}}<div class=trenner></div></td>
	</tr>{{/Fachbegriff}}{{#Norm}}
	<tr>
		<td class=anschlussfrage><font face="Arial"> ► </font>Norm</td>
	<tr>
		<td class=anschlussantwort>{{Norm}}<div class=trenner></div></td>
	</tr>{{/Norm}}{{#Klausurstandort}}
	<tr>
		<td class=anschlussfrage><font face="Arial"> ► </font>Klausurstandort</td>
	<tr>
		<td class=anschlussantwort>{{Klausurstandort}}<div class=trenner></div></td>
	</tr>{{/Klausurstandort}}

</table>

<div class=Quelle>{{Quellen}}</div>

</div>

<script>
function standard_gesetz_erkennen(){var deck="{{Deck}}"
if(deck.search(/Privatrecht|Zivilrecht/i)!=-1)
return"BGB"
else if(deck.search(/Strafrecht/i)!=-1)
return"StGB"
return""}
var injectScript=(src)=>{return new Promise((resolve,reject)=>{const script=document.createElement('script');script.src=src;script.async=!0;script.onload=resolve;script.onerror=reject;document.head.appendChild(script)})};setTimeout(function(){(async()=>{if(typeof normangaben_in_allen_relevanten_elementen_verlinken==='undefined'){await injectScript('_normen_verlinken.js')}
if(typeof normen_verlinkt=='undefined'){var normen_verlinkt=!0
normangaben_in_allen_relevanten_elementen_verlinken(standard_gesetz_erkennen())}})()},0)
</script>
```
  
</details>

**Comment by david-allison-1** [comment]
Thanks for the report. Duplicate of #8951. Fixed in 2.15.3, will be on the Play Store soon (hopefully today).

Best,

David

**Comment by david-allison-1** [comment]
Just to note; you can download 2.14.6 Parallel.A from here while the fix is pending release
https://github.com/ankidroid/Anki-Android/releases/tag/v2.14.6
